planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/test/test_utils.cpp

```diff
@@ -12,46 +12,1540 @@
+        .probability(1.0));
+    object_data.move_time = 0.4;
+
+    constexpr double margin = 0.0;
+    const auto pose =
+      path.points.at(autoware::motion_utils::findNearestIndex(path.points, object_pose.position))
+        .point.pose;
+
+    object_data.envelope_poly = createEnvelopePolygon(object_data, pose, margin);
+
+    EXPECT_TRUE(filtering_utils::isSatisfiedWithCommonCondition(
+      object_data, path, forward_detection_range, 6.6, createPoint(0.0, 0.0, 0.0), false,
+      parameters));
+  }
+
+  // within detection range.
+  {
+    const auto object_pose = geometry_msgs::build<geometry_msgs::msg::Pose>()
+                               .position(createPoint(4.5, 1.0, 0.0))
+                               .orientation(createQuaternionFromRPY(0.0, 0.0, deg2rad(45)));
+
+    ObjectData object_data;
+    object_data.object.shape.type = autoware_perception_msgs::msg::Shape::BOUNDING_BOX;
+    object_data.object.shape.dimensions.x = 2.8284271247461901;
+    object_data.object.shape.dimensions.y = 1.41421356237309505;
+    object_data.object.kinematics.initial_pose_with_covariance.pose = object_pose;
+    object_data.direction = Direction::LEFT;
+    object_data.object.classification.emplace_back(
+      autoware_perception_msgs::build<ObjectClassification>()
+        .label(ObjectClassification::TRUCK)
+        .probability(1.0));
+    object_data.move_time = 0.4;
+
+    constexpr double margin = 0.0;
+    const auto pose =
+      path.points.at(autoware::motion_utils::findNearestIndex(path.points, object_pose.position))
+        .point.pose;
+
+    object_data.envelope_poly = createEnvelopePolygon(object_data, pose, margin);
+
+    EXPECT_TRUE(filtering_utils::isSatisfiedWithCommonCondition(
+      object_data, path, forward_detection_range, 4.0, createPoint(0.0, 0.0, 0.0), true,
+      parameters));
+  }
+}
+
+TEST(TestUtils, isSameDirectionShift)
+{
+  constexpr double negative_shift_length = -1.0;
+  constexpr double positive_shift_length = 1.0;
+
+  {
+    ObjectData object;
+    object.direction = Direction::RIGHT;
+
+    ASSERT_TRUE(isSameDirectionShift(isOnRight(object), negative_shift_length));
+    ASSERT_FALSE(isSameDirectionShift(isOnRight(object), positive_shift_length));
+  }
+
+  {
+    ObjectData object;
+    object.direction = Direction::LEFT;
+    ASSERT_TRUE(isSameDirectionShift(isOnRight(object), positive_shift_length));
+    ASSERT_FALSE(isSameDirectionShift(isOnRight(object), negative_shift_length));
+  }
+
+  {
+    ObjectData object;
+    object.direction = Direction::NONE;
+    EXPECT_THROW(isSameDirectionShift(isOnRight(object), positive_shift_length), std::logic_error);
+  }
+}
+
+TEST(TestUtils, isShiftNecessary)
+{
+  constexpr double negative_shift_length = -1.0;
+  constexpr double positive_shift_length = 1.0;
+
+  {
+    ObjectData object;
+    object.direction = Direction::RIGHT;
+
+    ASSERT_TRUE(isShiftNecessary(isOnRight(object), positive_shift_length));
```

Comment:
This might be better to test all other tests.
```suggestion
    EXPECT_TRUE(isShiftNecessary(isOnRight(object), positive_shift_length));
```

Reply:
fix in https://github.com/autowarefoundation/autoware.universe/pull/9134/commits/34b64c36a5fa9a1a4c9972e355f1f9ce4b7c0cd6.